File: log-generators/test_consumer.py
from kafka import KafkaConsumer


#if __name__ == '__main__':
#    consumer = KafkaConsumer(
#        'raw',
#        group_id='test-consumer-group',
#        bootstrap_servers=['kafka-single-node:9092'],
#        auto_offset_reset='earliest',
#        enable_auto_commit=True,
#        # consumer_timeout_ms=1000
#    )
#
#    for message in consumer:
#        message = message.value.decode('utf8')
#        print(message)
#
#    while True:
#        messages = consumer.poll(timeout_ms=1000)
#        for message in messages:
#            print(message)

if __name__ == '__main__':
    consumer1 = KafkaConsumer(group_id='test-consumers',
                              bootstrap_servers=['kafka-single-node:9092'],
                              auto_offset_reset='earliest',
                              enable_auto_commit=True)
    consumer2 = KafkaConsumer(group_id='test-consumers',
                              bootstrap_servers=['kafka-single-node:9092'],
                              auto_offset_reset='earliest',
                              enable_auto_commit=True)

    consumer1.subscribe(['raw'])
    consumer2.subscribe(['raw'])

    while True:
        messages1 = consumer1.poll()
        for message in messages1:
            print("Consumer 1:", message.topic, message.partition, message.offset, message.key, message.value)

        messages2 = consumer2.poll()
        for message in messages2:
            print("Consumer 2:", message.topic, message.partition, message.offset, message.key, message.value)

File: src/java/TestConsumer.java
import java.time.Duration;
import java.util.Properties;
import java.util.Collections;

import org.apache.kafka.clients.consumer.KafkaConsumer;
import org.apache.kafka.clients.consumer.ConsumerConfig;
import org.apache.kafka.clients.consumer.ConsumerRecord;
import org.apache.kafka.clients.consumer.ConsumerRecords;
import org.apache.kafka.common.serialization.StringDeserializer;

public class TestConsumer {
    public static void main(String[] args) {
        // Set up the configuration for the Kafka consumer
        Properties config = new Properties();
        config.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, "kafka-single-node:9092");
        config.put(ConsumerConfig.GROUP_ID_CONFIG, "test-consumer-group");
        config.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
        config.put(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "true");
        config.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class.getName());
        config.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class.getName());

        // Create the Kafka consumer
        KafkaConsumer<String, String> consumer = new KafkaConsumer<>(config);

        // Subscribe to the 'raw' topic
        consumer.subscribe(Collections.singletonList("raw"));

        // Poll for new messages and process them
        try {
            while (true) {
                ConsumerRecords<String, String> records = consumer.poll(Duration.ofMillis(1000));
                for (ConsumerRecord<String, String> record : records) {
                    String key = record.key();
                    String value = record.value();
                    System.out.println("Key: " + key);
                    System.out.println("Value: " + value.substring(0, 90) + "...");
                }
            }
        } catch (Exception e) {
            System.out.println("error: " + e);
        } finally {
            consumer.close();
        }
    }
}
